packages/FormElement/src/hooks/useFormElement.js

```diff
@@ -0,0 +1,56 @@
+import React from "react";
+import nanoid from "nanoid";
+import isNil from "lodash/isNil";
+
+export default function useFormElement(props = { id: "", wrapperA11yProps: {} }) {
+  const { id, wrapperA11yProps } = props;
+  const wrapperAriaDescribedBy = wrapperA11yProps ? wrapperA11yProps["aria-describedby"] : "";
+  const ariaDescriptionId = React.useRef(nanoid()).current;
+  const ariaErrorId = React.useRef(nanoid()).current;
+  const ariaInstructionsId = React.useRef(nanoid()).current;
+  const uniqueInputId = React.useRef(nanoid()).current;
+
+  const ariaDescribedBy = `${ariaErrorId} ${ariaInstructionsId} ${ariaDescriptionId} ${wrapperAriaDescribedBy}`;
+  const generateLabelId = id => (isNil(id) || id === "" ? uniqueInputId : id);
+  const idForLabel = generateLabelId(id);
+  const refLabel = React.useRef(null);
+
+  return {
+    inputA11yProps: {
+      id: idForLabel,
+      "aria-describedby": ariaDescribedBy,
+    },
+    checkboxA11yProps: {
+      id: idForLabel,
+      "aria-describedby": ariaDescribedBy,
+    },
+    radioA11yProps: {
+      "aria-describedby": ariaDescribedBy,
+    },
+    datePickerA11yProps: {
+      id: idForLabel,
+      "aria-describedby": ariaDescribedBy,
+    },
+    listBoxA11yProps: {
+      refLabel,
+    },
```

Comment:
I don't like that this component would need to know about its consumer.  Can we just offer up all the props (`id`, `ariaDescribedBy`, `refLabel`) and let the consumer destructure the ones it needs?

Reply:
I can see that it is easier for a consumer to just get the spread the props they need without having to know which props they should be spreading, but I agree it does create a strange binding based on component name....
Personally I am more in favour of your suggested approach Mike but will wait to hear what others think.

Reply:
I second mike, we should strive for making the form element fully agnostic.